alpenhorn/auto_import.py

```diff
@@ -310,3 +261,73 @@ def on_deleted(self, event):
         if basename[0] == "." and basename[-5:] == ".lock":
             basename = basename[1:-5]
             import_file(self.node, self.root, os.path.join(dirname, basename))
+
+
+# Routines to control the filesystem watchdogs.
+# =============================================
+
+obs_list = None
+
+
+def setup_observers(node_list):
+    """Setup the watchdogs to look for new files in the nodes.
+    """
+
+    global obs_list
+
+    # If any node has auto_import set, look for new files and add them to the
+    # DB. Then set up a watchdog for it.
+    obs_list = []
+    for node in node_list:
+        if node.auto_import:
+
+            # Get list of all files that exist on the node
+            q = (ar.ArchiveFileCopy.select(ac.ArchiveFile.name, ac.ArchiveAcq.name)
+                 .where(ar.ArchiveFileCopy.node == node,
+                        ar.ArchiveFileCopy.has_file == 'Y')
+                 .join(ac.ArchiveFile).join(ac.ArchiveAcq))
+
+            already_imported_files = [os.path.join(a, f) for a, f in q.tuples()]
+
+            log.info("Crawling base directory \"%s\" for new files.", node.root)
+
+            for acq_name, d, f_list in os.walk(node.root):
```

Comment:
Good question. I thought this was tested by the docker tests, but let me check that one.